### PR TITLE
feat(opsgenie): Frontend for spike protection notifications

### PR DESCRIPTION
**Pull request by mifu67**
Add a notification action for Opsgenie.

https://github.com/getsentry/sentry/assets/83109586/07f17533-d0f0-4ce3-8bda-e43e5bdd8855

### CODE REVIEW

**Review by cathteng — COMMENTED** [review]
is there a difference between the Opsgenie form and the Pagerduty form? the logic seems identical except that you renamed services --> teams e.g. `getServiceOptions` --> `getTeamOptions`?

edit: if they're pretty similar i would rename the PagerdutyForm component to something that would indicate that it can be reused for both Pagerduty and Opsgenie

**Comment by mifu67** [comment]
> is there a difference between the Opsgenie form and the Pagerduty form? the logic seems identical except that you renamed services --> teams e.g. `getServiceOptions` --> `getTeamOptions`?
> 
> edit: if they're pretty similar i would rename the PagerdutyForm component to something that would indicate that it can be reused for both Pagerduty and Opsgenie

Will do this.

File: static/app/components/notificationActions/forms/onCallServiceForm.tsx [hunk 1]
```diff
@@ -5,51 +5,51 @@ import {Button} from 'sentry/components/button';
...
   AvailableNotificationAction,
   NotificationAction,
 } from 'sentry/types/notificationActions';
 
-type PagerdutyFormProps = {
+type OnCallServiceFormProps = {
+  /**
+   * Map of pagerduty/opsgenie integration IDs to available actions for those IDs
+   */
+  Integrations: Record<number, AvailableNotificationAction[]>;
```
Comment (cathteng) [inline static/app/components/notificationActions/forms/onCallServiceForm.tsx:19]: does this need to be capitalized?

File: static/app/components/notificationActions/forms/onCallServiceForm.tsx [hunk 2]
```diff
@@ -79,6 +79,13 @@ function PagerdutyForm({
       },
     }));
   };
+  const keySelectionTxt =
```
Comment (cathteng) [inline static/app/components/notificationActions/forms/onCallServiceForm.tsx:82]:
```suggestion
  const keySelectionText =
```

File: static/app/components/notificationActions/forms/onCallServiceForm.tsx [hunk 3]
```diff
@@ -79,6 +79,13 @@ function PagerdutyForm({
...
+    onCallService === 'pagerduty'
+      ? t('account with the service')
+      : t('account with the key');
+
+  const dropdownTxt =
```
Comment (cathteng) [inline static/app/components/notificationActions/forms/onCallServiceForm.tsx:87]:
```suggestion
  const dropdownText =
```

File: static/app/components/notificationActions/notificationActionManager.tsx [hunk 4]
```diff
@@ -170,15 +186,18 @@ function NotificationActionManager({
       ) {
         return;
       }
-
+      let label = t(
+        'Send a %s notification',
+        serviceType !== NotificationActionService.SENTRY_NOTIFICATION
+          ? capitalize(serviceType)
+          : 'Sentry'
+      );
```
Comment (cathteng) [inline static/app/components/notificationActions/notificationActionManager.tsx:194]: nit: you could also make 3 blocks in here checking the 3 possible conditions. i do think the way i originally wrote it isn't that clear

```typescript
const getLabel = () => {
  switch (serviceType) {
    case NotificationActionService.SENTRY_NOTIFICATION:
      return t('Send a Sentry notification');
    case NotificationActionService.OPSGENIE:
      return t('Send an Opsgenie notification');
    default:
      return t('Send a %s notification', capitalize(serviceType));
  }
}
```

File: static/app/components/notificationActions/forms/onCallServiceForm.tsx [hunk 5]
```diff
@@ -79,6 +79,13 @@ function PagerdutyForm({
...
+    onCallService === 'pagerduty' ? t('Select Service') : t('Select Key');
```
Comment (cathteng) [inline static/app/components/notificationActions/forms/onCallServiceForm.tsx:88]: what is a key and what is a team? i notice elsewhere that you also use team to refer to an opsgenie service. it should be consistent

**Review by cathteng — COMMENTED** [review]
this is looking good, thanks for the refactor! just a couple of small things left

File: static/app/components/notificationActions/forms/onCallServiceForm.tsx (hunk 5 above)
Comment (mifu67) [reply static/app/components/notificationActions/forms/onCallServiceForm.tsx:88]: I'll change this to team to be consistent

File: static/app/components/notificationActions/forms/onCallServiceForm.tsx [hunk 6]
```diff
@@ -14,7 +14,7 @@ import {
 
 type OnCallServiceFormProps = {
   /**
-   * Map of pagerduty/opsgenie integration IDs to available actions for those IDs
+   * Map of Pagerduty/Opsgenie integration IDs to available actions for those IDs
    */
   Integrations: Record<number, AvailableNotificationAction[]>;
```
Comment (cathteng) [inline static/app/components/notificationActions/forms/onCallServiceForm.tsx:19]: last thing: can you make this not capitalized?

**Review by cathteng — APPROVED** [review]
lgtm except for that last thing :)

i'm not sure if you need to merge the backend before this, though

**Pull request merged**